zerver/views/users.py

```diff
@@ -247,6 +249,15 @@ def avatar(
         # interact with fake email addresses anyway.
         if is_email:
             raise MissingAuthenticationError()
+
+        if settings.RATE_LIMITING:
+            try:
+                unique_avatar_key = str(realm.id) + email_or_id + "medium"
```

Comment:
This algorithm is unsound; user ID 1 in realm 21 will have the same key as user ID 12 in realm 1.

Let's use an f-string to assemble the key here, and use a separator (like `/`) between the distinct components.

Reply:
Sorry, fixed!